src/Microsoft.AspNet.Mvc.Core/FormatterMappings.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Net.Http.Headers;
+using System.Collections.ObjectModel;
+using System.Globalization;
+using Microsoft.AspNet.Mvc.Core;
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// These options are used to specify mapping between the Url Format and corresponding ContentType.
+    /// </summary>
+    public class FormatterMappings
+    {
+        private readonly Dictionary<string, MediaTypeHeaderValue> _map =
+            new Dictionary<string, MediaTypeHeaderValue>(StringComparer.OrdinalIgnoreCase);
+
+        public void SetFormatMapping([NotNull] string format, [NotNull] MediaTypeHeaderValue contentType)
+        {
+            if (contentType == null)
+            {
+                throw new ArgumentException((Resources.ArgumentCannotBeNullOrEmpty), "contentType");
+            }
+
+            format = RemovePeriodIfPresent(format);
+            _map[format] = contentType;
+        }
+
+        public MediaTypeHeaderValue GetContentTypeForFormat(string format)
```

Comment:
you have `SetFormatMapping` and `GetContentTypeForFormat`, which don't sound super-related. They should, since they are inverses of each other.